hack/tools/logcheck/main_test.go

```diff
@@ -23,5 +23,8 @@ import (
 )
 
 func TestAnalyzer(t *testing.T) {
+	// Migration test fails if correctness test fails.
+	// So unit testing with migration flag enabled is enough.
```

Comment:
Please add tests for different flag configuration.

Reply:
Added separate test data for testing different flag configuration.